src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -222,238 +209,53 @@
-          Object.prototype.hasOwnProperty.call(addonRelayAddress, "description")
-            ? addonRelayAddress.description
-            : "";
-
-        // Cache the generated_for alias attribute when updating local storage data.
-        // Note that this data attribute only exists in aliases generated through the add-on
-        let storedAliasGeneratedFor = addonRelayAddress?.generated_for ?? "";
-
-        // This covers any legacy label field and remaps them.
-        if (storedLegacyAliasLabel) {
-          storedAliasLabel = addonRelayAddress.domain;
-        }
+      if (localStorageRelayAddresses.length === 0 && existingLocalStorageRelayAddresses) {
+        localStorageRelayAddresses.push(...existingLocalStorageRelayAddresses);
+      }
 
-        // This covers any legacy siteOrigin field and remaps them.
-        const storedLegacyAliasSiteOrigin = addonRelayAddress?.siteOrigin;
-        if (storedLegacyAliasSiteOrigin) {
-          storedAliasGeneratedFor = addonRelayAddress.generated_for;
-        }
+      // Get data from server (Note: All description/generated_for fields should be BLANK)
+      const serverRelayAddresses = await apiRequest(apiRelayAddressesURL);
 
-        const aliasLabelForm = aliasCard.querySelector(
-          "form.relay-email-address-label-form"
-        );
-        const aliasLabelInput = aliasCard.querySelector(
-          "input.relay-email-address-label"
-        );
+      if (localStorageRelayAddresses.length === 0) {
+        // Nothing local! Start fresh! 
+        localStorageRelayAddresses.push(...serverRelayAddresses);
+        browser.storage.local.set({ relayAddresses: localStorageRelayAddresses });
+        return;
+      }
 
-        const aliasLabelWrapper = aliasLabelForm.parentElement;
-        aliasLabelWrapper.classList.add("show-label"); // Field is visible only to users who have the addon installed
+      // WIP/DON'T SHIP: This is a scrape method for the website to grab know label descriptions
+      // Actual fix should be to interacte with webpage local storage (however, localStorage does not have an labels stored currently) 
+      const masks = document.querySelectorAll("div[class*='Alias_controls']");
+      
+      masks.forEach(mask => {
+        const maskAddress = mask.querySelector("samp[class*='Alias_address']").textContent;
+        const maskDescriptionValue = mask.querySelector("input[class*='LabelEditor_label-input']").value;
 
-        aliasLabelInput.dataset.label = storedAliasLabel;
+        const filteredMask = localStorageRelayAddresses.filter(mask => mask.full_address === maskAddress);
 
-        if (storedAliasLabel !== "") {
-          aliasLabelInput.value = storedAliasLabel;
-          aliasLabelWrapper.classList.add("user-created-label");
-        } else {
-          aliasLabelInput.placeholder = defaultAliasLabelText;
+        if (filteredMask.length === 0) {
+          throw new Error("No match found between local data and scrapped page data");
```

Comment:
This case may not be possible, but I triggered it once during development so I wanted to catch it in case. Because parts of the add-on are relying on local storage to display masks, I want to make sure we're managing the object correctly.